components/automate-cli/cmd/chef-automate/ha_cert_show.go

```diff
@@ -0,0 +1,414 @@
+		c.writer.Println("\nPostgresql certificates are common across all nodes.\n")
+		c.printPublicAndPrivateKeys(certInfo.PostgresqlCertsByIP[0], "Postgresql", false)
+		return
+	}
+
+	for _, certs := range certInfo.PostgresqlCertsByIP {
+		c.printPublicAndPrivateKeys(certs, "Postgresql", true)
+	}
+
+}
+
+// printOpensearchCertificates prints the opensearch certificates
+func (c *certShowImpl) printOpensearchCertificates(certInfo certShowCertificates) {
+	c.writer.Title("Opensearch Certificates")
+	c.writer.HR()
+
+	c.writer.Println("=======================Opensearch Root CA=======================")
+	if len(strings.TrimSpace(certInfo.OpensearchRootCert)) == 0 {
+		c.writer.Println("No Opensearch root certificate found")
+	} else {
+		c.writer.Println(certInfo.OpensearchRootCert)
+	}
+
+	c.writer.Println("\n======================Opensearch Admin Key======================")
+	if len(strings.TrimSpace(certInfo.OpensearchAdminKey)) == 0 {
+		c.writer.Println("No admin key found")
+	} else {
+		c.writer.Println(certInfo.OpensearchAdminKey)
+	}
+
+	c.writer.Println("\n======================Opensearch Admin Cert======================")
+	if len(strings.TrimSpace(certInfo.OpensearchAdminCert)) == 0 {
+		c.writer.Println("No admin certificate found")
+	} else {
+		c.writer.Println(certInfo.OpensearchAdminCert)
+	}
+
+	if len(certInfo.OpensearchCertsByIP) == 0 {
+		c.writer.Println("No public and private key found for Opensearch")
+		return
+	}
+
+	if c.flags.node == "" && c.isCommonCerts(certInfo.OpensearchCertsByIP) {
+		c.writer.Println("\nOpensearch certificates are common across all nodes.\n")
+		c.printPublicAndPrivateKeys(certInfo.OpensearchCertsByIP[0], "Opensearch", false)
+		return
+	}
+
+	for _, certs := range certInfo.OpensearchCertsByIP {
+		c.printPublicAndPrivateKeys(certs, "Opensearch", true)
+	}
+
+}
+
+// printPublicAndPrivateKeys prints the public and private keys for a given service
+func (c *certShowImpl) printPublicAndPrivateKeys(certs CertByIP, remoteService string, printNode bool) {
+	if c.flags.node != "" && c.flags.node != certs.IP {
+		return
+	}
+
+	if printNode {
+		c.writer.Println(fmt.Sprintf("\n%s Certificates for %s\n", remoteService, certs.IP))
+	}
+
+	c.writer.Println(fmt.Sprintf("=======================%s Public Key=======================", remoteService))
+	if len(strings.TrimSpace(certs.PublicKey)) == 0 {
+		c.writer.Println(fmt.Sprintf("No %s public key found", remoteService))
+	} else {
+		c.writer.Println(certs.PublicKey)
+	}
+
+	c.writer.Println(fmt.Sprintf("\n======================%s Private Key======================", remoteService))
+	if len(strings.TrimSpace(certs.PrivateKey)) == 0 {
+		c.writer.Println(fmt.Sprintf("No %s private key found", remoteService))
+	} else {
+		c.writer.Println(certs.PrivateKey)
+	}
+}
+
+// validateNode validates if the node exists in the cluster
+func (c *certShowImpl) validateNode(certs []CertByIP, remoteService string) error {
```

Comment:
is this function to validate the given IP belongs to the cluster?
If so we already implemented that in cert-rotate, can we use that?

Reply:
We cannot use it directly as implementation differs based on the signature. So we have to modify the original function in order to use it.